### ISSUE TITLE
Reusing HttpRequest

### ISSUE THREAD

**Issue by mathie** [igrigorik/em-http-request#77]
I ran into a problem yesterday trying to reuse a single `HttpRequest` for multiple requests. (The idea was that the server would support keepalive/pipelining, but sadly it turned out not to be the case!). Here's the gist of what I was trying to do:

```
EM.run do
  request = EventMachine::HttpRequest.new('http://api.example.com/')
  deferrable = request.get :path => '/api/v2/chickens.json'
  deferrable.callback do
    deferred_eggs = request.get :path => '/api/v2/chickens/1/eggs.json'
    deferred_eggs.callback { puts "It worked"; EM.stop }
    deferred_eggs.errback  { puts "Egg retrieval failed"; EM.stop }
  end
  deferrable.errback  { puts "It failed"; EM.stop }
end
```

The second request would appear to be issued, but would never complete. I wrote up some detail on how I went about debugging it here: http://woss.name/2011/03/06/using-tcpflow/

The short version is that I think, if the server responded with `Connection: close` it's closing the TCP connection, and that a subsequent request should `#reconnect` before submitting another request. I think. But I'm finding it a little difficult to figure out where that should happen as I've not used EventMachine before, and it's been a _long_ time since I last used Twisted!

If you can point me in the right direction for fixing it, I'll give it a go and feed back a pull request.
